### PR TITLE
Fix attention scaling factor

### PR DESCRIPTION
**Pull request by TianyiQ**
In line with both the original paper and pytorch's [implementation](https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L6362), the scaling factor / normalizing factor on the attention matrix should be 1/sqrt(head_size) instead of 1/sqrt(embedding_size).

### CODE REVIEW

**Comment by FareedKhan-dev** [comment]
Hey @TianyiQ, I accidentally reverted everything, including your intended commit (c118aaf). I've now reapplied it separately, so the correct fix is back in place. Thank you!

Reference: [Commit c118aaf](https://github.com/FareedKhan-dev/train-llm-from-scratch/commit/9acf1f3a8b1839fa7e183fe7040adee506586c12)
